src/modeldata.jl

```diff
@@ -32,36 +32,36 @@ function Base.read(io::IOStream, ::Type{Config})
     Config(dim, hidden_dim, n_layers, n_heads, n_kv_heads, vocab_size, seq_len, shared_weights)
 end
 
-@kwdef struct TransformerWeights{T}
+@kwdef struct TransformerWeights{T, Vec <: AbstractVector{T}, Mat <: AbstractMatrix{T}, Arr <: AbstractArray{T, 3}}
```

Comment:
there might be a nicer way to do this... Not sure

Reply:
I'm fine with this change; have been meaning to include more exotic matrix types. Would prefer a name like `Arr3` instead of `Arr` to make the 3-d-ness explicit.

Reply:
Good point, switched to `Arr3`